React/Modules/RCTTiming.m

```diff
@@ -163,6 +176,11 @@ - (void)startTimers
   }
 }
 
+- (BOOL)hasPendingTimers
+{
+  return _sendIdleEvents || _timers.count > 0;
+}
+
 - (void)didUpdateFrame:(__unused RCTFrameUpdate *)update
```

Comment:
I think this is what we want in this case, requestIdleCallback is called at the end of every frame that has some time left with the frame start time. Then JS can schedule work knowing how much time it has left.